apps/studio/src/components/common/texteditor/TextEditor.vue

```diff
@@ -0,0 +1,335 @@
+            shortcut: this.ctrlOrCmd("shift+z"),
+          },
+          {
+            name: "Cut",
+            slug: "",
+            handler: () => {
+              const selection = this.editor.getSelection();
+              this.editor.replaceSelection("");
+              this.$native.clipboard.writeText(selection);
+            },
+            class: selectionDepClass,
+            shortcut: this.ctrlOrCmd("x"),
+          },
+          {
+            name: "Copy",
+            slug: "",
+            handler: () => {
+              const selection = this.editor.getSelection();
+              this.$native.clipboard.writeText(selection);
+            },
+            class: selectionDepClass,
+            shortcut: this.ctrlOrCmd("c"),
+          },
+          {
+            name: "Paste",
+            slug: "",
+            handler: () => {
+              const clipboard = this.$native.clipboard.readText();
+              if (this.editor.getSelection()) {
+                this.editor.replaceSelection(clipboard, "around");
+              } else {
+                const cursor = this.editor.getCursor();
+                this.editor.replaceRange(clipboard, cursor);
+              }
+            },
+            shortcut: this.ctrlOrCmd("v"),
+          },
+          {
+            name: "Delete",
+            slug: "",
+            handler: () => {
+              this.editor.replaceSelection("");
+            },
+            class: selectionDepClass,
+          },
+          {
+            name: "Select All",
+            slug: "",
+            handler: () => {
+              this.editor.execCommand("selectAll");
+            },
+            shortcut: this.ctrlOrCmd("a"),
+          },
+          {
+            type: "divider",
+          },
+          {
+            name: "Find",
+            slug: "find",
+            handler: () => {
+              this.editor.execCommand("find");
+            },
+            shortcut: this.ctrlOrCmd("f"),
+          },
+          {
+            name: "Replace",
+            slug: "replace",
+            handler: () => {
+              this.editor.execCommand("replace");
+            },
+            shortcut: this.ctrlOrCmd("r"),
+          },
+          {
+            name: "ReplaceAll",
+            slug: "replace_all",
+            handler: () => {
+              this.editor.execCommand("replaceAll");
+            },
+            shortcut: this.ctrlOrCmd("shift+r"),
+          },
+        ],
+        event,
+      };
+
+      const customOptions = this.contextMenuOptions
+        ? this.contextMenuOptions(event, menu.options)
+        : undefined;
+
+      if (customOptions === false) {
+        return;
+      }
+
+      if (customOptions === undefined) {
+        this.$bks.openMenu(menu);
+      } else {
+        this.$bks.openMenu({
+          ...menu,
+          options: customOptions,
+        });
+      }
+    },
+  },
+  mounted() {
+    this.initialize();
+    this.$emit("interface", {
```

Comment:
We shouldn't be leaking objects with methods in events like this. It needs to only be data we send upstream with props. This really breaks the communication model of Vue.